### PR TITLE
IT - WDB - 2532- Test wazuh-DB getconfig WDB command

### PR DESCRIPTION
**Pull request by CamiRomero**
In this PR the test suite for `wazuhdb getconfig` command is added. A total of 5 new test cases covering different modes and combinations of options.

<table>
  <tr>
    <th>Related issues</th>
    <th><a href="[url](https://github.com/wazuh/wazuh-qa/issues/2532)">#2532</a></th>
    <th><a href="[url](https://github.com/wazuh/wazuh/issues/10771)">#10771 (Wazuh)</a></th>
  </tr>
</table>

## Cases for `wazuh-DB getconfig` command:

- [x] Wazuh-DB getconfig commands
        - Get config missing parameter
        - Get config empty parameter
        - Get config wrong parameter
        - Get internal config
        - Get wdb config

--- 

## Configuration options
|OS | Version | Branch | Type |
|--|--|--|--|
| CentOS | 4.4.0 | dev-10771-agent-groups-files-to-wazuh-db | Manager|

**Local Internal Options:** No Local Internal Options used

---

---

## Tests Results
| Test Path | Os/Type | Execution Type | Results | Date | By |
|--|--|--|--|--|--|
| test_wazuh_db/test_wazuhdb_getconfig  | CentOS - Manager | Local | [:green_circle:]((https://github.com/wazuh/wazuh-qa/files/8173375/r1.zip))[:green_circle:](https://github.com/wazuh/wazuh-qa/files/8173376/R2.zip)[:green_circle:](https://github.com/wazuh/wazuh-qa/files/8173377/R3.zip)| 02/03/2022 |  @CamiRomero  |  
| test_wazuh_db/test_wazuhdb_getconfig   | CentOS - Manager | Jenkins | [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/22206/)  [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/22207/)  [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/22208/) |02/03/2022 |  @CamiRomero |

---
## Required
- [x] Proven that tests **pass** when they have to pass.
- [x] Proven that tests **fail** when they have to fail.
- [x] Python codebase satisfies PEP-8 style style guide. `pycodestyle --max-line-length=120 --show-source --show-pep8 file.py`.
- [x] Python codebase is documented following the Google Style for Python docstrings.
- [x] The test is documented in wazuh-qa/docs.

### CODE REVIEW

**Review by damarisg — CHANGES_REQUESTED** [review]

File: tests/integration/test_wazuh_db/test_wazuhdb_getconfig.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        - Verify that the socket response matches the expected output.
+    input_description:
+        - Test cases are defined in the wazuhdb_getconfig.yaml file.
+    expected_output:
+        - an array with the configuration of DB.
+    tags:
+        - wazuh_db
+        - wdb_socket
+    '''
+    case_data = test_case[0]
```
[the remark this answers is not in the document]
Comment (CamiRomero) [reply tests/integration/test_wazuh_db/test_wazuhdb_getconfig.py:97]: Done in commit [6c4e30](https://github.com/wazuh/wazuh-qa/pull/2627/commits/6c4e30d3bed6eded11c0f190239da054faafe368)

File: tests/integration/test_wazuh_db/test_wazuhdb_getconfig.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+# Marks
+pytestmark = [pytest.mark.linux, pytest.mark.tier(level=0), pytest.mark.server]
+
+# Configurations
+test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+messages_file = os.path.join(os.path.join(test_data_path, 'global'), 'wazuhdb_getconfig.yaml')
+module_tests = []
+with open(messages_file) as f:
+    module_tests.append((yaml.safe_load(f), messages_file.split('_')[0]))
```
[the remark this answers is not in the document]
Comment (CamiRomero) [reply tests/integration/test_wazuh_db/test_wazuhdb_getconfig.py]: Done in commit [a3c13](https://github.com/wazuh/wazuh-qa/pull/2627/commits/a3c13eb0c30c6df3183ad9eea26e0a11b86c7420)

File: deps/wazuh_testing/wazuh_testing/tools/file.py [hunk 3]
```diff
@@ -51,6 +51,22 @@ def read_yaml(file_path):
...
+    """Read a YAML file from a given path, return a list with the YAML data
+    after apply filter
+
+    Args:
+        file_path (str): Path of the YAML file to be readed
+        filter (str): filder to extract some part of yaml
+        list: a list with previous information
+
+    Returns:
+       list
```
Comment (damarisg) [inline deps/wazuh_testing/wazuh_testing/tools/file.py:64]: specifies what this list contains with its type.

**Review by damarisg — CHANGES_REQUESTED** [review]

File: deps/wazuh_testing/wazuh_testing/tools/file.py (hunk 3 above)
Comment (CamiRomero) [reply deps/wazuh_testing/wazuh_testing/tools/file.py]: Done in commit [12919](https://github.com/wazuh/wazuh-qa/pull/2627/commits/12919d62cd044a989fb0e2781e1ee9145c3b33e4)

**Review by damarisg — APPROVED** [review]
## Tests Results
| Test Path | Os/Type | Execution Type | Results | Date | By |
|--|--|--|--|--|--|
| test_wazuh_db/test_wazuhdb_getconfig | CentOS/Manager | Jenkins | [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/22380/) [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/22373/) [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/22371/) |2022/03/04|  Seyla

**Comment by damarisg** [comment]
GJ Cami!!

**Pull request merged**